File: PharmaDicBackEnd/PharmaDicBackEnd.ApiService/DTOs/MedicineDetailDto.cs
namespace PharmaDicBackEnd.ApiService.DTOs
{
    public class MedicineDetailDto
    {
        public int MedicineId { get; set; }
        public string MedicineName { get; set; } = null!;
        public string? CategoryName { get; set; }
        public string? DosageForm { get; set; }
        public string? Manufacturer { get; set; }
        public string? Uses { get; set; }
        public string? Dosage { get; set; }
        public string? Contraindications { get; set; }
        public string? SideEffects { get; set; }

        
        public List<IngredientDto> Ingredients { get; set; } = new List<IngredientDto>();
    }

    public class IngredientDto
    {
        public string IngredientName { get; set; } = null!;
        public string? Amount { get; set; } // Hàm lượng (VD: 500mg)
    }
}

File: PharmaDicMobile/app/src/main/java/com/example/pharmadicmobile/dtos/MedicineDetailDto.java
package com.example.pharmadicmobile.dtos;

import java.util.ArrayList;
import java.util.List;

public class MedicineDetailDto {
    private int medicineId;
    private String medicineName;
    private String categoryName;
    private String dosageForm;
    private String manufacturer;
    private String uses;
    private String dosage;
    private String contraindications;
    private String sideEffects;
    private List<IngredientDto> ingredients = new ArrayList<>();

    public MedicineDetailDto() {}

    public int getMedicineId() { return medicineId; }
    public void setMedicineId(int medicineId) { this.medicineId = medicineId; }

    public String getMedicineName() { return medicineName; }
    public void setMedicineName(String medicineName) { this.medicineName = medicineName; }

    public String getCategoryName() { return categoryName; }
    public void setCategoryName(String categoryName) { this.categoryName = categoryName; }

    public String getDosageForm() { return dosageForm; }
    public void setDosageForm(String dosageForm) { this.dosageForm = dosageForm; }

    public String getManufacturer() { return manufacturer; }
    public void setManufacturer(String manufacturer) { this.manufacturer = manufacturer; }

    public String getUses() { return uses; }
    public void setUses(String uses) { this.uses = uses; }

    public String getDosage() { return dosage; }
    public void setDosage(String dosage) { this.dosage = dosage; }

    public String getContraindications() { return contraindications; }
    public void setContraindications(String contraindications) { this.contraindications = contraindications; }

    public String getSideEffects() { return sideEffects; }
    public void setSideEffects(String sideEffects) { this.sideEffects = sideEffects; }

    public List<IngredientDto> getIngredients() { return ingredients; }
    public void setIngredients(List<IngredientDto> ingredients) { this.ingredients = ingredients; }
}
